bin/adhoc-scripts/updatePileupObjects.py

```diff
@@ -0,0 +1,185 @@
+        raise RuntimeError(msg)
+
+    # decode and load JSON data
+    sdata = resp.read().decode('utf8')
+    data = json.loads(sdata)
+
+    logger.debug("Response: %s", data)
+    if data and not data.get("result", []):
+        msg = f"Failed to retrieve pileup documents. Error: {data}"
+        raise RuntimeError(msg)
+    return data["result"]
+
+
+def writePileupDocs(mspileupUrl, puDocs, certDict, logger):
+    """
+    Update each pileup document in MSPileup
+    :param mspileupUrl: string with the MSPileup url
+    :param puDocs: list with pileup documents
+    :param certDict: dictionary with cert and key reference
+    :param logger: logger object
+    :return: None
+    """
+    url = mspileupUrl + "/ms-pileup/data/pileup"
+    headers = {'Content-Type': 'application/json'}
+    logger.info("Starting to update pileup documents against backend: %s", url)
+    cafile = certDict['cert']
+    capath = '/etc/grid-security/certificates'
+    for doc in puDocs:
+        logger.info("Updating pileup document: %s", doc.get("pileupName"))
+        req = urllib.request.Request(url)
+        req.add_header('Accept', 'application/json')
+        context = getContext(certDict['cert'], certDict['key'], certDict['cafile'])
+        resp = urllib.request.urlopen(req, context=context)
+        sdata = resp.read().decode('utf8')
+        data = json.loads(sdata)
+
+        logger.debug("Response: %s", data)
+        if data and data.get("result", []):
+            msg = f"Failed to inject pileup document for {doc['pileupName']}. Error: {data}"
+            logger.critical(msg)
+
+
+def main():
+    """Executes everything"""
+    optmgr = OptionParser()
+    opts = optmgr.parser.parse_args()
+    logger = logging.getLogger('updatePileupObjects')
+    logger.setLevel(logging.INFO)
+    logging.basicConfig()
+
+    # setup proxy/cert
+    proxy = os.getenv('X509_USER_PROXY', '')
+    if proxy:
+        cert = proxy
+        key = proxy
+    else:
+        cert = os.getenv('X509_USER_CERT', '')
+        key = os.getenv('X509_USER_KEY', '')
+        if not cert or not key:
+            defaultCert = os.path.join(os.getenv('HOME'), '.globus/usercert.pem')
+            defaultKey = os.path.join(os.getenv('HOME'), '.globus/userkey.pem')
+            if os.path.isfile(defaultCert) and os.path.isfile(defaultKey):
+                cert = defaultCert
+                key = defaultKey
+    if cert and key:
+        hdict = {'cert': cert, 'key': key}
+    cafile = opts.cafile
+    if not cafile:
+        cafile = '/etc/grid-security/certificates/CERN-Root-2.pem'
+    if not os.path.isfile(cafile):
+        logger.error("Please locate CERN CA file and provide it via --cafile option. Current cafile='%s' does not exist", cafile)
+        sys.exit(1)
+    hdict['cafile'] = cafile
+    logger.info("Using %s and %s", cert, key)
+
+    # check that user either provide cert or proxy
+    if not cert and not proxy:
```

Comment:
I would say this block can be removed, once you add the else clause above (see comment above).

Reply:
code re-organized already